graphql/src/graphql_schema.ml

```diff
@@ -476,6 +464,67 @@ module Make (Io : IO) (Stream: Stream with type 'a io = 'a Io.t) = struct
     coerce = fun x -> `String x;
   }
 
+  (* Mandatory directives: skip and include *)
+  let skip_directive = Directive {
+    name = "skip";
+    doc = Some "Directs the executor to skip this field or fragment when the `if` argument is true.";
+    locations = [`Field; `Fragment_spread; `Inline_fragment];
+    typ = non_null bool;
+    args = Arg.[
+      arg "if" ~doc:"Skipped when true." ~typ:(non_null bool)
+    ];
+    resolve = fun if_ -> if_
```

Comment:
Maybe return `not if_` here to make the directives homogenous in the semantics of their return value (i.e. whether the field should be included or not)?

Reply:
Interesting thought – I think that conflates the directive definition with its usage in `should_include`, though. 

I'm imagining that we could want the raw value of `"skip"` in some future usage (i.e. if the spec evolves).

Reply:
What if the return type was ```[`Skip | `Include]```?